File: poker_server/poker-card-server.js
const http = require('http');
const crypto = require('crypto');
const url = require('url');
const querystring = require('querystring');

class AdvancedRandomGenerator {
    constructor() {
        this.reseed();
    }

    reseed() {
        const seedSources = [
            crypto.randomBytes(4).readUInt32BE(0),
            Date.now(),
            crypto.randomBytes(4).readUInt32BE(0),
            this.hashCode(JSON.stringify(this.state))
        ];
        const seed = crypto.createHash('sha512').update(seedSources.toString()).digest();
        this.state = new Uint32Array(seed.buffer, 0, 4);
    }

    hashCode(str) {
        let hash = 0;
        for (let i = 0; i < str.length; i++) {
            const char = str.charCodeAt(i);
            hash = ((hash << 5) - hash) + char;
            hash = hash & hash; // Convert to 32-bit integer
        }
        return hash;
    }

    generateNumbers(minValue, maxValue, count) {
        const extra = Math.max(100, count);
        const numbers = new Array(count + extra).fill(0).map(() => 
            Math.floor(this.state[0] / 0x100000000 * (maxValue - minValue + 1) + minValue)
        );

        for (let i = numbers.length - 1; i > 0; i--) {
            const j = Math.floor(this.state[0] / 0x100000000 * (i + 1));
            [numbers[i], numbers[j]] = [numbers[j], numbers[i]];
        }

        const result = numbers.slice(0, count);
        this.reseed();
        return result;
    }
}

class MersenneTwister {
    constructor(seed = null) {
        this.w = 32;
        this.n = 624;
        this.m = 397;
        this.r = 31;
        this.a = 0x9908B0DF;
        this.u = 11;
        this.d = 0xFFFFFFFF;
        this.s = 7;
        this.b = 0x9D2C5680;
        this.t = 15;
        this.c = 0xEFC60000;
        this.l = 18;
        this.f = 1812433253;

        this.MT = new Array(this.n).fill(0);
        this.index = this.n + 1;
        this.lowerMask = (1 << this.r) - 1;
        this.upperMask = ~this.lowerMask >>> 0;

        this.seedMt(seed || Date.now());
    }

    seedMt(seed) {
        this.MT[0] = seed >>> 0;
        for (let i = 1; i < this.n; i++) {
            this.MT[i] = (this.f * (this.MT[i-1] ^ (this.MT[i-1] >>> (this.w-2))) + i) >>> 0;
        }
        this.twist();
    }

    extractNumber() {
        if (this.index >= this.n) {
            this.twist();
        }

        let y = this.MT[this.index];
        y = y ^ ((y >>> this.u) & this.d);
        y = y ^ ((y << this.s) & this.b);
        y = y ^ ((y << this.t) & this.c);
        y = y ^ (y >>> this.l);

        this.index++;
        return y >>> 0;
    }

    twist() {
        for (let i = 0; i < this.n; i++) {
            const x = (this.MT[i] & this.upperMask) + (this.MT[(i+1) % this.n] & this.lowerMask);
            let xA = x >>> 1;
            if (x % 2 !== 0) {
                xA = xA ^ this.a;
            }
            this.MT[i] = this.MT[(i + this.m) % this.n] ^ xA;
        }
        this.index = 0;
    }

    randint(a, b) {
        const rangeSize = b - a + 1;
        if (rangeSize <= 0) {
            throw new Error("Invalid range");
        }
        
        const bitsNeeded = (rangeSize - 1).toString(2).length;
        const mask = (1 << bitsNeeded) - 1;
        
        while (true) {
            const value = this.extractNumber() & mask;
            if (value < rangeSize) {
                return a + value;
            }
        }
    }
}

const advancedRng = new AdvancedRandomGenerator();
const mersenneTwister = new MersenneTwister(advancedRng.generateNumbers(0, 2**32 - 1, 1)[0]);
const suits = ['hearts', 'diamonds', 'clubs', 'spades'];
const values = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A'];

const server = http.createServer((req, res) => {
    const parsedUrl = url.parse(req.url);
    
    if (req.method === 'OPTIONS') {
        res.writeHead(200, {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'GET, OPTIONS',
            'Access-Control-Allow-Headers': 'X-Requested-With'
        });
        res.end();
        return;
    }

    if (parsedUrl.pathname === '/draw') {
        const query = querystring.parse(parsedUrl.query);
        const numCards = parseInt(query.count) || 5;

        const cardIndices = Array.from({length: numCards}, () => mersenneTwister.randint(0, 51));

        const cards = cardIndices.map(index => ({
            suit: suits[Math.floor(index / 13)],
            value: values[index % 13]
        }));

        const response = { cards };

        res.writeHead(200, {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*'
        });
        res.end(JSON.stringify(response));
    } else {
        res.writeHead(404);
        res.end();
    }
});

const PORT = 3040;
server.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});

File: poker_server/poker-card-server.py
import secrets
from http.server import HTTPServer, BaseHTTPRequestHandler
import json
import time
import hashlib
import os
import random
import numpy as np
from urllib.parse import urlparse, parse_qs

class AdvancedRandomGenerator:
    def __init__(self):
        self.numpy_rng = np.random.RandomState()
        self.reseed()

    def reseed(self):
        seed_sources = [
            int.from_bytes(os.urandom(4), 'big'),
            int(time.time() * 1000000),
            int.from_bytes(secrets.token_bytes(4), 'big'),
            hash(str(self.numpy_rng.get_state()))
        ]
        seed = hashlib.sha512(str(seed_sources).encode()).digest()
        seed_int = int.from_bytes(seed[:4], 'big')
        self.numpy_rng.seed(seed_int)

    def generate_numbers(self, min_value, max_value, count):
        extra = max(100, count)
        numbers = self.numpy_rng.randint(min_value, max_value + 1, size=count + extra)
        
        for i in range(len(numbers) - 1, 0, -1):
            j = self.numpy_rng.randint(0, i + 1)
            numbers[i], numbers[j] = numbers[j], numbers[i]
        
        result = numbers[:count].tolist()
        self.reseed()
        return result

class MersenneTwister:
    def __init__(self, seed=None):
        self.w, self.n, self.m, self.r = 32, 624, 397, 31
        self.a = 0x9908B0DF
        self.u, self.d = 11, 0xFFFFFFFF
        self.s, self.b = 7, 0x9D2C5680
        self.t, self.c = 15, 0xEFC60000
        self.l = 18
        self.f = 1812433253

        self.MT = [0] * self.n
        self.index = self.n + 1
        self.lower_mask = (1 << self.r) - 1
        self.upper_mask = ~self.lower_mask

        if seed is None:
            seed = int(time.time())
        self.seed_mt(seed)

    def seed_mt(self, seed):
        self.MT[0] = seed
        for i in range(1, self.n):
            self.MT[i] = (self.f * (self.MT[i-1] ^ (self.MT[i-1] >> (self.w-2))) + i) & 0xffffffff
        self.twist()

    def extract_number(self):
        if self.index >= self.n:
            self.twist()

        y = self.MT[self.index]
        y = y ^ ((y >> self.u) & self.d)
        y = y ^ ((y << self.s) & self.b)
        y = y ^ ((y << self.t) & self.c)
        y = y ^ (y >> self.l)

        self.index += 1
        return y & 0xffffffff

    def twist(self):
        for i in range(self.n):
            x = (self.MT[i] & self.upper_mask) + (self.MT[(i+1) % self.n] & self.lower_mask)
            xA = x >> 1
            if x % 2 != 0:
                xA = xA ^ self.a
            self.MT[i] = self.MT[(i + self.m) % self.n] ^ xA
        self.index = 0

    def randint(self, a, b):
        range_size = b - a + 1
        if range_size <= 0:
            raise ValueError("Invalid range")
        
        bits_needed = (range_size - 1).bit_length()
        mask = (1 << bits_needed) - 1
        
        while True:
            value = self.extract_number() & mask
            if value < range_size:
                return a + value

class PokerCardHandler(BaseHTTPRequestHandler):
    advanced_rng = AdvancedRandomGenerator()
    mersenne_twister = MersenneTwister(advanced_rng.generate_numbers(0, 2**32 - 1, 1)[0])
    suits = ['hearts', 'diamonds', 'clubs', 'spades']
    values = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A']

    def do_GET(self):
        parsed_path = urlparse(self.path)
        if parsed_path.path == '/draw':
            query_components = parse_qs(parsed_path.query)
            num_cards = int(query_components.get('count', [5])[0])  # Default to 5 if not specified
            
            card_indices = [self.mersenne_twister.randint(0, 51) for _ in range(num_cards)]
            
            cards = []
            for index in card_indices:
                suit = self.suits[index // 13]
                value = self.values[index % 13]
                cards.append({'suit': suit, 'value': value})
            
            response = {'cards': cards}
            
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps(response).encode())
        else:
            self.send_error(404)

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'X-Requested-With')
        self.end_headers()

def run_server(port=3040):
    server_address = ('', port)
    httpd = HTTPServer(server_address, PokerCardHandler)
    print(f'Server running on port {port}')
    httpd.serve_forever()

if __name__ == '__main__':
    run_server()